src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1155,6 +1169,14 @@ void TaskSketcherElements::on_settings_extendedInformation_changed()
     slotElementsChanged();
 }
 
+void TaskSketcherElements::on_settings_autoCollapseFilter_changed()
+{
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Elements");
+    hGrp->SetBool("AutoCollapseFilter", ui->settingsButton->actions()[1]->isChecked());

```

Comment:
Use `ui->settingsButton->actions()[1]->isChecked()` only once and store in a scoped variable.

Reply:
Done